std/bigint.d

```diff
@@ -1025,6 +1025,220 @@ public:
         assert(aa[BigInt(456)] == "def");
     }
 
+    /*
+        Increments an arbitrary-length integer, represented as an array of
+        bytes, using the native endianness.
+
+        This is just used by `toBytes` to convert unsigned bytes of the
+        BigInt to two's complement form.
+    */
+    private static void incrementBytes (ref ubyte[] value) nothrow pure @safe
+    {
+        if (value.length == 0u)
+        {
+            value = [ 0x01u ];
+            return;
+        }
+
+        version (BigEndian)
+        {
+            // This loop adds one to an arbitrary length array of bytes.
+            for (size_t i = (value.length - 1); i < 0u; i--)
+            {
+                if (value[i] != 0xFFu)
+                {
+                    value[i]++;
+                    break;
+                }
+
+                value[i] = 0x00u;
+
+                // If the array of bytes is maxed out, append a next byte set to one.
+                if (i == (value.length - 1))
+                {
+                    value = (0x01u ~ value);
+                    return;
+                }
+            }
+        }
+        else version (LittleEndian)
+        {
+            // This loop adds one to an arbitrary length array of bytes.
+            for (size_t i = 0u; i < value.length; i++)
+            {
+                if (value[i] != 0xFFu)
+                {
+                    value[i]++;
+                    break;
+                }
+
+                value[i] = 0x00u;
+
+                // If the array of bytes is maxed out, append a next byte set to one.
+                if (i == (value.length - 1))
+                {
+                    value ~= 0x01u;
+                    return;
+                }
+            }
+        }
+    }
+
+    @safe
+    unittest
+    {
+        ubyte[] a = [ 0x00u ];
+        incrementBytes(a);
+        assert(a == [ 0x01u ]);
+
+        ubyte[] b = [ 0xFFu ];
+        incrementBytes(b);
+        assert(b == [ 0x00u, 0x01u ]);
+
+        ubyte[] c = [ 0xFDu, 0xFFu, 0xFFu ];
+        incrementBytes(c);
+        assert(c == [ 0xFEu, 0xFFu, 0xFFu ]);
+    }
+
+    /**
+        Converts the `BigInt` to a sequence of bytes that represents the
+        native-endian representation of that number.
+    */
+    ubyte[] toBytes() const @system @property nothrow
+    in
+    {
+        assert(this.uintLength > 0u);
+    }
+    out (ret)
+    {
+        assert(ret.length > 0u);
+    }
+    body
+    {
+        ubyte[] ret;
```

Comment:
Use an `appender!(ubyte[])` i think

Reply:
Well, I subsequently set the length of the array to the length that will be needed, then dump the `uint`s in that array. How would `Appender` improve that?

Reply:
okay, forget the appender thing.